shell/renderer/api/electron_api_blink_utils.cc

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2023 Salesforce, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/renderer/api/electron_api_blink_utils.h"
+
+#include "shell/common/gin_helper/dictionary.h"
+#include "shell/common/node_includes.h"
+#include "third_party/blink/public/web/web_blob.h"
+
+namespace electron {
+
+namespace api {
+
+namespace blink_utils {
+
+std::string GetPathForFile(gin_helper::Arguments* args,
+                           v8::Local<v8::Value> file) {
+  blink::WebBlob blob = blink::WebBlob::FromV8Value(file);
+  if (blob.IsNull()) {
+    args->ThrowError(
+        "getPathForFile expected to receive a File object but one was not "
+        "provided");
+    return "";
+  }
+  return blob.Path();
+}
+
+}  // namespace blink_utils
+
+}  // namespace api
+
+}  // namespace electron
```

Comment:
Nest that namespace bad boy.

```suggestion
namespace electron::api::blink_utils {

std::string GetPathForFile(gin_helper::Arguments* args,
                           v8::Local<v8::Value> file) {
  blink::WebBlob blob = blink::WebBlob::FromV8Value(file);
  if (blob.IsNull()) {
    args->ThrowError(
        "getPathForFile expected to receive a File object but one was not "
        "provided");
    return "";
  }
  return blob.Path();
}

}  // namespace electron::api::blink_utils
```

Reply:
This is the pattern we use elsewhere, not sure why, I assume it's in the chromium styleguide 🤷

Reply:
🤔 You used the nested namespaces in `shell/renderer/api/electron_api_blink_utils.h` though?

We should be using nested namespaces everywhere after #37855.

Reply:
Weird, idk how, I copied these from template files 🤔 Will clean it up

Reply:
I don't see anything in the style guide, but `rg -t h namespace | rg ::` indicates that  upstream is fine with the terser form of declaring nested namespaces :smile_cat:, e.g.

```sh
extensions/browser/browsertest_util.h:namespace extensions::browsertest_util {
extensions/browser/browsertest_util.h:}  // namespace extensions::browsertest_util
extensions/browser/extension_function_histogram_value.h:namespace extensions::functions {
extensions/browser/extension_function_histogram_value.h:}  // namespace extensions::functions
extensions/browser/api/extensions_api_client.h:      std::map<settings_namespace::Namespace, ValueStoreCache*>* caches);
extensions/browser/api/app_runtime/app_runtime_api.h:namespace api::app_runtime {
extensions/browser/api/app_runtime/app_runtime_api.h:}  // namespace api::app_runtime
```

Reply:
@ckerr, it's [squirreled away in the "Modern C++ Usage in Chromium" part of their style guide](https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++-features.md#nested-namespaces-allowed), with a link to their discussion thread, where the TL;DR is they approved it for usage but didn't feel the need to go do a mass cleanup of existing code, just use it going forward on new code.